build_for_docssite.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+#
+# This script prepares the README for use in the docs site. The main thing it does is remove the title and replace it
+# with a frontmatter.
+#
+
+function create_artifact {
+  local -r target_file="$1"
+  local -r target_dir="$(dirname $target_file)"
+
+  mkdir -p "$target_dir"
+  cp ./README.md "$target_file"
+}
+
+function replace_header {
+  local -r infile="$1"
+
+  local -r frontmatter='---\
+title: "Gruntwork IaC Library Catalog"\
+date: __DATE__\
+tags: ["terraform"]\
```

Comment:
Should this data be read in from `gruntydocs.yml` rather than hardcoded in the script?

Reply:
Or perhaps given the likely temporary nature of this repo, it doesn't matter for now.

Reply:
There are two reasons for this:

One reason is that the frontmatter needs to be injected into the markdown files shipped to the `doc-sourcer`. I am trying to keep the login in the `doc-sourcer` really dumb (run script && copy output). `gruntydocs.yml` is actually the config for the `doc-sourcer`, so this is why it isn't in there, because it isn't the responsibility of the `doc-sourcer` to inject the frontmatter: it is the responsibility of the doc compiler for the repo (in this case, this script). This flexibility in the sourcer is what allows for more complex compile chains like the `boilerplate` based prototype I showed before.

The second reason is that I couldn't for the life of me get it to work when I was pulling in this data from a file. My bash-fu was too weak to figure out what combination of newline escape sequences would make it work with `sed`...